lib/scripts/controllers/options.js

```diff
@@ -97,6 +99,7 @@ export class OptionCtrl {
       this.blacklistedWords = data.userBlacklistedWords.slice(1, -1).split('|');
       this.blacklistedWebsites = data.blacklist.slice(1, -1).split('|');
       this.userDefinedTranslations = JSON.parse(data.userDefinedTranslations);
+      this.userDefinedTranslationList = this.toList(this.userDefinedTranslations);
```

Comment:
Hi @ankit-m . I'm not convinced. Storing the same information twice does make the code more prone to bugs. Suppose that we are storing X and Y (in a database or as public variables in a class), and Y is supposed to be always just a transformation of X. In other words, it is supposed to be the case that Y = f(X). You implement the code in a careful way, guaranteeing that this condition always holds. But nothing prevents a future developer from not noticing that this condition should be enforced and implementing something that will update, let's say, only Y, changing it to Y'. Then, suddenly our database or our public variables have Y' and X stored, and Y' != f(X).

The usual good way to deal with this is to store either X or Y only.  Then, whenever you need the other, you just convert it. For example, if you stored X and you need to use Y, you just compute f(X). If you stored Y and you need X, you just compute f^{-1}(Y).

An exception to this rule is when the conversion function "f" is too slow, but I don't think this is an issue in our case. (Or is it?)

Could you consider this and see if you can improve the code correspondingly?